### ISSUE TITLE
Strange behaviour when selecting UTF-8 chars

### ISSUE THREAD

**Issue by BBasile** [Alexey-T/ATSynEdit#104]
### How to reproduce

- start an application that contains an editor (e.g demo_simplest)
- paste the following sample `"ຂອ້ຍກິນແກ້ວໄດ້ໂດຍທີ່ມັນບໍໄດ້ເຮັດໃຫ້ຂອ້ຍເຈັບ"`
- select the line that contains the sample, select till end <kbd>SHIFT</kbd>+<kbd>END</kbd>.
- reduce step by step the selection <kbd>SHIFT</kbd>+<kbd>LEFT</kbd>.
- You'll see strange characters:

![1](https://cloud.githubusercontent.com/assets/7805501/8893953/e3d411c2-33a4-11e5-954e-8026a2b9a58a.png)
![2](https://cloud.githubusercontent.com/assets/7805501/8893956/e8aae130-33a4-11e5-88e2-cfd7e9a901fb.png)

here the accent is stripped from the char and put to its right. It looks like the selection split the two bytes of a character.

### Note
Despite of the bug, the behaviour with UTF-8 is a bit better than in SynEdit ([**cf this problem in Laz forums**](http://forum.lazarus.freepascal.org/index.php/topic,26483.msg162792.html#msg162792)). Currently I'm trying to see if it would be a big win to use ATSynEdit instead of SynEdit.

Do you think it can be fixed ?

**Comment by Alexey-T** [comment]
maybe can be, See. U have 2 widechars: hat and accept. when rendering all string, accent rendered over hat, when rendered variant 2, hat is rendered at own place, and accent with "circle" rendered on 2nd place

What to do?

**Comment by Alexey-T** [comment]
Linux even more worse: unicode renderer is weird

![uni-strange](https://cloud.githubusercontent.com/assets/4154978/8894112/a4b81740-33b3-11e5-9bc9-62e233f82739.png)

accent rendered bad.

**Comment by BBasile** [comment]
I've tried to find an OSS editor that handle the case correctly and so far it's a failure.
- Notepadd++, so scintilla, NO.
- MonoDevelop, so probably the Gtk source editor, NO.

Among non OSS editors I've found:
- Powershell ISE: OK

But this won't help much...
One interesting thing is that web browser editors generally work fine:
- https://stackedit.io/editor: OK
- https://dpaste.de/: OK
- http://pastebin.com/: OK

However since I'm not in web development at all i can't say which **stuff.js** is used to render unicode text.
